etest_test/unit_test/tests_test.py

```diff
@@ -179,26 +167,19 @@ def setUp(self):
 
         type(self.mocked_overlay_overlay).ebuilds = unittest.mock.PropertyMock(return_value=ebuilds)
 
-        self._patch("Test")
-
-    mocks.add("overlay")
-
-    @helpers_test.mock("overlay")
-    def mock_overlay(self):
-        """Mock overlay."""
-        self._patch("overlay")
-        self.mocked_overlay_overlay = unittest.mock.MagicMock()
-        self.mocked_overlay.Overlay.return_value = self.mocked_overlay_overlay
+        patcher = unittest.mock.patch.object(sut, "Test")
+        self.mocked_Test = patcher.start()
+        self.addCleanup(patcher.stop)
 
     def test_tests(self):
         """tests.Tests()."""
-        self.tests = tests.Tests()
+        self.tests = sut.Tests()
         self.assertEqual(self.tests.ebuild_selector, [])
         self.assertEqual(len(list(self.tests)), len(self.test_calls))
         self.assertEqual(self.mocked_Test.mock_calls, self.test_calls)
 
     def test_tests_with_filter_with_version(self):
         """tests.Tests(('etest-9999.ebuild',))."""
-        self.tests = tests.Tests(("etest-9999.ebuild",))
+        self.tests = sut.Tests(("etest-9999.ebuild",))
```

Comment:
how would this react to ebuilds with no -9999 versions?

Reply:
Good question.  That sounds like a test case we could add once I make a bit more progress on cleaning things up (or if you're feeling brave, you could try adding in the current test suite).